auth/service.hh

```diff
@@ -69,6 +70,13 @@ public:
     using std::invalid_argument::invalid_argument;
 };
 
+/// A description of a CQL command from which auth::service can tell whether or not this command could endanger
+/// internal data on which auth::service depends.
+struct command_desc {};
+
+/// A predicate on command_desc that decides whether it's safe to execute.
+using protector = bool(*)(command_desc);
```

Comment:
Shouldn't that be std::function (or noncopyable_function)? So it can contain state?

In the bikeshed department, "protector" doesn't indicate what the bool signifies. Is it protected or allowed? Either return a bool_class, or change the name to is_protected or is_allowed.

Reply:
Abandoned protectors.